common: main.c: make show_boot_progress __weak

Back port upstream fix for newer compiler version.  Original commit
follows:

commit 3422299dc28fa8257677d03cc1253e3c9bf17e9f
Author: Jeroen Hofstee <jeroen@myspectrum.nl>
Date:   Thu Jun 26 20:18:31 2014 +0200

    common: main.c: make show_boot_progress __weak

    This not only looks a bit better it also prevents a warning with
    W=1 (no previous prototype).

    Signed-off-by: Jeroen Hofstee <jeroen@myspectrum.nl>
    Acked-by: Simon Glass <sjg@chromium.org>

diff --git a/common/main.c b/common/main.c
index 5d8454e..95cd29a 100644
--- a/common/main.c
+++ b/common/main.c
@@ -58,8 +58,7 @@ DECLARE_GLOBAL_DATA_PTR;
 /*
  * Board-specific Platform code can reimplement show_boot_progress () if needed
  */
-void inline __show_boot_progress (int val) {}
-void show_boot_progress (int val) __attribute__((weak, alias("__show_boot_progress")));
+__weak void show_boot_progress(int val) {}
 
 #if defined(CONFIG_UPDATE_TFTP)
 int update_tftp (ulong addr);
